### PR TITLE
[CQ-4348387] added changes for formatters and related javascript dynamic handling in edit dialogs for text,number and date-picker.

### PR DESCRIPTION
**Pull request by deepprakash345**
Verified the working of edit dialog of date-picker, number input, text input component. Verified that display pattern dropdown and display format textbox behave dynamically in parity to AF1.

Runtime of Display pattern and display picture clause was not working. Need to have a separate jira for runtime handling for patterns/formatters validation handling.
cc:- @nit23uec , @vdua

### CODE REVIEW

**Comment by nit23uec** [comment]
we should have a unit test case here for the sling model.

File: bundles/af-core/src/test/resources/form/formcontainer/datasource/test-content.json [hunk 1]
```diff
@@ -20,5 +20,14 @@
                 "jcr:primaryType": "nt:unstructured"
             }
         }
+    },
+    "formattertypedatasource": {
+        "jcr:primaryType"   : "nt:unstructured",
+        "datasource" : {
+            "sling:resourceType": "core/fd/components/form/container/v1/datasource",
+            "type": "formatters",
+            "guideDataModel": "basic",
```
Comment (nit23uec) [inline bundles/af-core/src/test/resources/form/formcontainer/datasource/test-content.json:29]: do we need guideDataModel? we shouldn;t ideally unless there's a bug

Comment (deepprakash345) [reply bundles/af-core/src/test/resources/form/formcontainer/datasource/test-content.json]: removed the,

**Comment by deepprakash345** [comment]
@nit23uec , can you please review this.

File: ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/textinput/v1/textinput/_cq_dialog/.content.xml [hunk 2]
```diff
@@ -291,7 +286,63 @@
...
+                                            name="./editPatternType">
+                                        <datasource
+                                                jcr:primaryType="nt:unstructured"
+                                                sling:resourceType="core/fd/components/form/container/v1/datasource"
+                                                fieldType="text-input"
+                                                type="formatters"/>
+                                    </editPatternType>
+                                    <editFormat
+                                            jcr:primaryType="nt:unstructured"
+                                            granite:class="cmp-adaptiveform-textinput__editformat"
```
Comment (nit23uec) [inline ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/textinput/v1/textinput/_cq_dialog/.content.xml:337]: what are the options being listed here for text?

Comment (deepprakash345) [reply ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/textinput/v1/textinput/_cq_dialog/.content.xml:337]: Hi @nitin, this works as it used to work in af1. On the basis of the option selected in Patterntype dropdown the corresponding regex gets filled in this editFormat textField. If the author makes some changes in editFormat then the patternType dropdown changes to "Custom".

**Comment by nit23uec** [comment]
- we have to remove picture clause options from text-input formatters list (with value as text{*})
- in case only one option is there in the type for text-input formatters, we should hide the type field and just show regex field.
- default value of formatters for date input and number inputs should be empty (so that no value is saved no opening and saving dialog without interacting with formatters tab)
- apply margins to the tabs of the edit dialog

**Comment by deepprakash345** [comment]
Hi @nitin,
Made the proposed changes, 
As far as the 2nd option mentioned by you, since we have now select option as default in the dropdown in parity with AF1, i think we should let the dropdown visible and the user should select custom to type the regex.

File: bundles/af-core/src/test/java/com/adobe/cq/forms/core/components/internal/servlets/FormMetaDataDataSourceServletTest.java [hunk 3]
```diff
@@ -94,6 +101,74 @@ public void testDataSource() throws Exception {
...
+
+    @Test
+    public void testDataSourceForTextInputFormatters() throws Exception {
+        ArrayList<FormsManager.ComponentDescription> componentDescriptions = new ArrayList<>();
+        componentDescriptions.add(formatters);
+
+        Resource mockedResource = Mockito.mock(Resource.class);
+        ValueMap mockedValueMap = new MockValueMap(mockedResource);
+        mockedValueMap.put("pattern1", "pat=text{999-99-9999}");
+        mockedValueMap.put("abc", "cat=text{999-99-9999}");
```
Comment (nit23uec) [inline bundles/af-core/src/test/java/com/adobe/cq/forms/core/components/internal/servlets/FormMetaDataDataSourceServletTest.java:146]: how are we verifying that pictureclause is getting filtered out from text-input formatters

**Comment by deepprakash345** [comment]
@nit23uec , please review this pr, made the suggested changes

**Comment by deepprakash345** [comment]
Hi @nit23uec , can you please review this

File: ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/textinput/v1/textinput/_cq_dialog/.content.xml (hunk 2 above)
Comment (nit23uec) [reply ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/textinput/v1/textinput/_cq_dialog/.content.xml:338]: the display and edit patterns should not be listed for text-input.

**Comment by deepprakash345** [comment]
@nit23uec , made the changes and mentioned the things covered in this PR in a comment in JIRA https://jira.corp.adobe.com/browse/CQ-4348387.

File: ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/textinput/v1/textinput/_cq_dialog/.content.xml [hunk 4]
```diff
@@ -267,32 +267,33 @@
...
-                                    <validatePictureClause
+                                    <validateFormat
                                             jcr:primaryType="nt:unstructured"
+                                            granite:class="cmp-adaptiveform-textinput__validationformat"
                                             sling:resourceType="granite/ui/components/coral/foundation/form/textfield"
                                             fieldDescription="Regular Expression is permitted."
                                             fieldLabel="Pattern"
-                                            name="./pattern">
-                                    </validatePictureClause>
+                                            name="./validateFormat">
```
Comment (nit23uec) [inline ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/textinput/v1/textinput/_cq_dialog/.content.xml:286]: the name should be `pattern` as per spec. any reason why it's changed to validateFormat i the latest commit?

File: ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/textinput/v1/textinput/_cq_dialog/.content.xml [hunk 5]
```diff
@@ -267,32 +267,33 @@
...
+                                    </validateFormat>
                                     <validatePictureClauseMessage
                                             jcr:primaryType="nt:unstructured"
                                             sling:resourceType="granite/ui/components/coral/foundation/form/textarea"
                                             fieldLabel="Error Message"
-                                            name="./validatePictureClauseMessage"/>
+                                            name="./validatePatternMessage"/>
```
Comment (nit23uec) [inline ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/textinput/v1/textinput/_cq_dialog/.content.xml:292]: sling models still read validatePictureClauseMessage  
See: https://github.com/adobe/aem-core-forms-components/blob/master/bundles/af-core/src/main/java/com/adobe/cq/forms/core/components/util/AbstractBaseImpl.java#L332

File: ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/textinput/v1/textinput/_cq_dialog/.content.xml (hunk 4 above)
Comment (deepprakash345) [reply ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/textinput/v1/textinput/_cq_dialog/.content.xml:286]: this was a miss at my end, i was going through the migration spec to check what should be the name to save this field, i made it in the same format in displayformat,editFormat etc. Should have validated in Sling model.

File: ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/textinput/v1/textinput/_cq_dialog/.content.xml (hunk 5 above)
Comment (deepprakash345) [reply ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/textinput/v1/textinput/_cq_dialog/.content.xml:292]: reverted this

File: ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/textinput/v1/textinput/clientlibs/editor/js/editDialog.js [hunk 6]
```diff
@@ -25,6 +25,12 @@
         TEXTINPUT_RICHTEXTVALUE = EDIT_DIALOG + " .cmp-adaptiveform-textinput__richtextvalue",
         TEXTINPUT_AUTOCOMPLETE = EDIT_DIALOG + " .cmp-adaptiveform-textinput__autocomplete",
         TEXTINPUT_AUTOFILL_FIELD_KEYWORD = EDIT_DIALOG + " .cmp-adaptiveform-textinput__autofillfieldkeyword",
+        TEXTINPUT_DISPLAYPATTERN = EDIT_DIALOG + " .cmp-adaptiveform-textinput__displaypattern",
+        TEXTINPUT_DISPLAYFORMAT = EDIT_DIALOG + " .cmp-adaptiveform-textinput__displayformat",
+        TEXTINPUT_EDITPATTERN = EDIT_DIALOG + " .cmp-adaptiveform-textinput__editpattern",
```
Comment (nit23uec) [inline ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/textinput/v1/textinput/clientlibs/editor/js/editDialog.js:30]: lets clean this up. we do not require most of this.

File: ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/textinput/v1/textinput/clientlibs/editor/js/editDialog.js [hunk 7]
```diff
@@ -67,6 +73,30 @@
             changeFormFields(["./_value", "./_richTextValue@Delete"], [filteredValue, null]);
         }
     }
-    Utils.initializeEditDialog(EDIT_DIALOG)(handleRichText);
+    function handleDisplayPatternDropDown(dialog) {
+        Utils.handlePatternDropDown(dialog,TEXTINPUT_DISPLAYPATTERN,TEXTINPUT_DISPLAYFORMAT);
+    }
+
+    function handleDisplayFormat(dialog){
```
Comment (nit23uec) [inline ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/textinput/v1/textinput/clientlibs/editor/js/editDialog.js:80]: edit and display format code not required.

File: ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/textinput/v1/textinput/clientlibs/editor/js/editDialog.js (hunk 6 above)
Comment (deepprakash345) [reply ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/textinput/v1/textinput/clientlibs/editor/js/editDialog.js]: done

**Review by nit23uec — APPROVED** [review]

**Comment by deepprakash345** [comment]
@nit23uec , can we merge this now

**Pull request merged**